### ISSUE TITLE
Module not found: Error: Cannot resolve module 'module' in [...]source-map-support @ source-map-support.js 464:17-34

### ISSUE THREAD

**Issue by mperdikeas** [evanw/node-source-map-support#155]
I am using `source-map-support` `0.4.5` in a module of mine (let's call it `moduleA`).

I then publish `moduleA` to npm and all's well till that point.

Then I try to use that `moduleA` in another project of mine (let's call it `moduleUser`). After I do `npm install` in `moduleUser` I try to do an `npm run build` which is configured in `package.json` as:

```
$ grep \"build\"\: package.json 
    "build": "webpack --progress --colors",
```

&hellip; and I then encounter (at the end of the build process) the following message which I can't decipher:

ERROR in ../~/moduleA/~/source-map-support/source-map-support.js
Module not found: Error: Cannot resolve module 'module' in /home/carsten/moduleUser/node_modules/moduleA/node_modules/source-map-support
 @ ../~/moduleA/~/source-map-support/source-map-support.js 464:17-34

Looking in the identified location in file `source-map-support.js` the line is:

```
var Module = require('module');
```

**Comment by mmmeff** [comment]
Also started receiving this same error on our production builds.

**Comment by jknight12882** [comment]
Receiving this as well

**Comment by mmmeff** [comment]
You can circumvent it by adding source-map-support@0.4.3 to your
project.json. It'll force the resolution for babel-register and avoid the
latent issues in 0.4.4+

**Comment by mperdikeas** [comment]
Having `^0.4.3` in my `package.json`:

    $ cat package.json | grep \"source-map-support\"
        "source-map-support": "^0.4.3"

&hellip; actually fetches `0.4.5`:

    $ npm ls --depth=0 | grep source-map-support
    └── source-map-support@0.4.5

&hellip; and the same problem persists after npm-publishing the package and using it from another project.

**Comment by mmmeff** [comment]
@mperdikeas So remove the `^` from `^0.4.3` ...

**Comment by mperdikeas** [comment]
Nailing the `source-map-support` version to `0.4.3` indeed fixes the problem.

**Comment by tomxtobin** [comment]
I've been seeing the same issue, and likewise ended up pinning to `0.4.3`. I'm also using Webpack for my builds, for what that's worth.

**Comment by jknight12882** [comment]
Any chance this is going to get fixed? Stuck on `0.4.3` until this is resolved

**Comment by Levino** [comment]
Same issue here.

**Comment by LinusU** [comment]
Hmm, I thought that we fixed this, is still a problem in **0.4.8**?

**Comment by veeramarni** [comment]
I still see this issue in 0.4.11.

**Comment by searsaw** [comment]
Still seeing it in `0.4.17`

**Comment by pmunin** [comment]
Still seeing in 0.5.0 :-( Is anyone from contributors looking into it?

**Comment by Ventzy** [comment]
I am using Angular 5, Angular cli 1.5.2 and everything was fine until I added and used 
`
import * as ts from "typescript";
`

My error starts with
```
WARNING in ./node_modules/source-map-support/source-map-support.js
Module not found: Error: Can't resolve 'module' in '/usr/local/app/node_modules/source-map-support'
resolve 'module' in '/usr/local/app/node_modules/source-map-support'
  Parsed request is a module
  using description file: /usr/local/app/node_modules/source-map-support/package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: /usr/local/app/node_modules/source-map-support/package.json (relative path: .)
    resolve as module
      /usr/local/app/node_modules/source-map-support/node_modules doesn't exist or is not a directory
      /usr/local/app/node_modules/node_modules doesn't exist or is not a directory
      /usr/local/node_modules doesn't exist or is not a directory
      /usr/node_modules doesn't exist or is not a directory
      /node_modules doesn't exist or is not a directory
      looking for modules in /usr/local/app/node_modules
        using description file: /usr/local/app/package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
        after using description file: /usr/local/app/package.json (relative path: ./node_modules)
          using description file: /usr/local/app/package.json (relative path: ./node_modules/module)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/app/node_modules/module doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/app/node_modules/module.ts doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/app/node_modules/module.js doesn't exist
            as directory
              /usr/local/app/node_modules/module doesn't exist

```
And continues something like 1000 more lines...  Not sure what I can do about it. Any idea?
Here is my package.json.
```
{
  "name": "app",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve --host=0.0.0.0 --port=80 --disable-host-check --poll=500",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^5.1.0",
    "@angular/cdk": "^5.0.0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/material": "^5.0.0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/router": "^5.0.0",
    "angular-new-router": "github:angular/router",
    "angular2-draggable": "^1.0.7",
    "core-js": "^2.4.1",
    "hammerjs": "^2.0.8",
    "immutable": "^3.8.2",
    "jquery": "^3.2.1",
    "jquery-ui": "^1.12.1",
    "monaco-editor": "^0.10.1",
    "ng-drag-drop": "^4.0.0",
    "rxjs": "^5.5.2",
    "zone.js": "^0.8.14"
  },
  "devDependencies": {
    "@angular/cli": "1.5.2",
    "@angular/compiler-cli": "^5.0.0",
    "@angular/language-service": "^5.0.0",
    "@types/jasmine": "~2.5.53",
    "@types/jasminewd2": "~2.0.2",
    "@types/jquery": "^3.2.16",
    "@types/jqueryui": "^1.11.37",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.2.0",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.2.0",
    "tslint": "~5.7.0",
    "typescript": "~2.4.2"
  }
}
```

**Comment by noahgibbs** [comment]
Seeing the same issue in 0.5.1, as well as unresolved 'fs' module. The following addition to the top level of my webpack.config.js fixes it:

    node: {
        fs: "empty",
        module: "empty"
     }

**Comment by Vadorequest** [comment]
I have a similar issue:
```
next These dependencies were not found:
next 
next * fs in ./node_modules/source-map-support/source-map-support.js
next * module in ./node_modules/source-map-support/source-map-support.js
next 
next To install them, you can run: npm install --save fs module
```

For anyone having the issue with the Next.js library, here is the workaround:

`next.config.js`
```js
module.exports = {
  webpack: (config, { buildId, dev, isServer, defaultLoaders }) => {
    config.node = {
      fs: 'empty',
      module: "empty",
    };
    return config;
  },
};
```

https://github.com/webpack-contrib/css-loader/issues/447#issuecomment-368345121

**Comment by sunny920406** [comment]
attention

**Comment by sunny920406** [comment]
I still see this issue in 0.5.4.

**Comment by alundiak** [comment]
I have this issue with `v0.4.18`. Downgraded to `v0.4.3` and issue still exists.
But I have 
```
+-- babel-core@6.26.0
| `-- babel-register@6.26.0
|   `-- source-map-support@0.4.18
```
And I think babel-core uses 0.4.18 no matter of source-map-support installed package on top level.
I didn't have this issue long time, because in my React/Webpack codebase I didn't ahve require('module') untill now. And I rather need it it for some time, which later on myabe will be removed/moved/modified/deprecated. 

But the issue interesting. AMD vs. CJS.

But then I changed my webpack.config.js and added one more line:
```
node: {
		fs: 'empty',
		module: 'empty'
}
```

And issue gone.

So far so good for me, but such behavior is odd :) Just imagine, 2018 year, we work on NPM modules, module is essential for require/CJS and we have to add some explicit code to "see" it :)

**Comment by ideal-it-solutions** [comment]
I don't know what am doing wrong please someone help take a look at my config and how i tried to apply the workaround.

`var Encore = require('@symfony/webpack-encore');

Encore

  // the project directory where compiled assets will be stored
    .setOutputPath('public/build/')
    // the public path used by the web server to access the previous directory
    .setPublicPath('/build')
    .cleanupOutputBeforeBuild()
    .enableSourceMaps(!Encore.isProduction())
    // uncomment to create hashed filenames (e.g. app.abc123.css)
    //.enableVersioning(Encore.isProduction())

    // uncomment to define the assets of the project

    .addEntry('public', './assets/js/public.js')
    .addEntry('member', './assets/js/member.js')
    .addEntry('admin', './assets/js/admin.js')

     //.addStyleEntry('css/app', './assets/css/app.scss')
    .addStyleEntry('global', './assets/css/global.scss')

    .createSharedEntry('vendor', [
        'jquery'
       // 'bootstrap'

        // you can also extract CSS - this will create a 'vendor.css' file
        // this CSS will *not* be included in page1.css or page2.css anymore
      // 'bootstrap-sass/assets/stylesheets/_bootstrap.scss'
    ])


    // uncomment if you use Sass/SCSS files
     .enableSassLoader(function(sassOptions) {}, {
             resolveUrlLoader: false
     })
    // 'fonts'and  images will be inlined
   /* .configureUrlLoader({
        fonts: { limit: 4096 },
        images: { limit: 4096 }
    })
*/
   /*trying to add the work around here and i get error*/
    .node({
        fs:'empty'
    })
     // show OS notifications when builds finish/fail
    .enableBuildNotifications()

    // uncomment for legacy applications that require $/jQuery as a global variable
     .autoProvidejQuery()



;

module.exports = Encore.getWebpackConfig();`

when i run 
`yarn run encore dev
`
i get error
` Error: Encore.node is not a recognized property or method.

  - index.js:942 Object.get
    [chukolo]/[@symfony]/webpack-encore/index.js:942:27

  - webpack.config.js:44 Object.<anonymous>
    C:/xampp/htdocs/chukolo/webpack.config.js:44:5

  - module.js:652 Module._compile
    module.js:652:30

  - module.js:663 Object.Module._extensions..js
    module.js:663:10

  - module.js:565 Module.load
    module.js:565:32

  - module.js:505 tryModuleLoad
    module.js:505:12

  - module.js:497 Function.Module._load
    module.js:497:3

  - module.js:596 Module.require
    module.js:596:17

  - module.js:11 require
    internal/module.js:11:18

  - convert-argv.js:97 requireConfig
    [chukolo]/[webpack]/bin/convert-argv.js:97:18


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this comm
and.
`

**Comment by leaphawk** [comment]
Is there any solution for this problem yet? facing the same thing.

**Comment by ideal-it-solutions** [comment]
> 
> 
> Is there any solution for this problem yet? facing the same thing.

What worked for me was deleting my instance of yarn and webencore then reinstalling it. sorry for the late reply

**Comment by ideal-it-solutions** [comment]
can you confirm that this (source-map-support' )was installed to your node_modules dir ?
iff you are using npm  https://docs.npmjs.com/cli/install
 or 
yarn add package name.
confirm the package name by checking https://yarnpkg.com/en/

**Comment by nogo10** [comment]
I get same error ```Module not found: Error: Can't resolve 'module' in ..node_modules/source-map-support ``` in versions 0.5.16, 0.5.18, and 0.4.18 (versions used in my packages)

**Comment by LinusU** [comment]
@mperdikeas, @nogo10, and anyone else in this thread; could you try #273 and see if that fixes the problem for you?

**Comment by nogo10** [comment]
To test it I copy pasted source-map-support.js code fix at #273 to replace instances in my app packages that use source-map-support.

Seems to work: I no longer get the ```Module not found ...``` error
(I do still get an unrelated scrypt error 
```.../node_modules/web3-eth-accounts/src/scrypt.js  Critical dependency: the request of a dependency is an expression  ```)

**Comment by LinusU** [comment]
Fixed in version 0.5.19 🚀

**Comment by patricktheoptimist** [comment]
To fix this I noticed that sourcemap was installed under myproject/node_modules/css_loader/node_modules/source-map/source-map.js

I grabbed the file, renamed it sourceMaps.js and saved it here: myproject/node_modules/css_loader/dist/runtime/sourceMaps.js

That fixed it. My guess is that the sourceMaps file has the same [logic ](https://developer.chrome.com/blog/sourcemaps/) in most cases, but is setup in a different directory now than in the past. I'm guessing that the structure changed for how the node_modules subfolders get setup within css-loader. Probably older versions still assume that the necessary file lives at /dist/runtime/sourceMaps.js

Probably there's a way to remap where the project is looking for sourceMaps.js and what it is called, but I find that copy and pasting  a file is much simpler and seems appropriately mundane.
